### DISCUSSION TITLE
Remote Schemas with UUID primary key: only fields with arguments or subfields can be toggled

### DISCUSSION THREAD

**Discussion by opslock-benjamin** [hasura/graphql-engine#7338] [Question]
```graphql
type Subtask {
  uuid: ID!
  task_uuid: ID!
  title: String!
}

type Query {
  subtasks: [Subtask!]!
}

input NewSubtask {
  task_uuid: ID!
  title: String!
}

type Mutation {
  createSubTask(input: NewSubtask!): Subtask!
}
```

This is what GQLGEN schema looks like.

I can successfully run my local GraphQL server , run mutations and queries.

I have a Hasura table called Tasks which has a `uuid` primary key.

When I try to add a remote relationship from Tasks -> (remote schema).Subtasks

I do not have the option to link `subtasks.task_uuid` to `tasks.uuid`

![image](https://user-images.githubusercontent.com/85521853/127381597-a114413e-6e28-4a80-8625-6cd896e9ccb5.png)

What do I need to modify in my schema so that Hasura allows me to associate `subtasks.task_uuid` to `task.uuid`?

**Comment by 0x777** [comment]
@opslock-benjamin graphql-engine will require a field on the remote schema with argument(s) to satisfy the join condition. In the above example, you'll nee the `subtasks` field to accept a `task_id` argument, so that graphql engine can associate a task with its subtasks. 
```graphql
type Query {
  subtasks(taskId: ID!): [Subtask!]!
}

```
